code/iaas/logic/src/main/java/io/cattle/platform/process/containerevent/ContainerEventCreate.java

```diff
@@ -0,0 +1,207 @@
+            public void doWithLockNoResult() {
+                // TODO INDEX ON EXTERNAL ID
+                Instance instance = objectManager.findAny(Instance.class, INSTANCE.ACCOUNT_ID, accountId,
+                        INSTANCE.EXTERNAL_ID, externalId);
+                if ( instance == null ) {
+                    instance = objectManager.newRecord(Instance.class);
+                    instance.setKind(InstanceConstants.KIND_CONTAINER);
+                    instance.setAccountId(accountId);
+                    instance.setExternalId(externalId);
+                    setName(inspect, instance);
+                    setNetwork(accountId, inspect, instance);
+                    setImage(inspect, instance);
+                    setHost(event, instance);
+                    DataAccessor.fields(instance).withKey(InstanceConstants.FIELD_NATIVE_CONTAINER).set(Boolean.TRUE);
+                    instance = objectManager.create(instance);
+                }
+
+                if ( !InstanceConstants.STATE_RUNNING.equals(ObjectUtils.getState(instance)) &&
+                        !InstanceConstants.STATE_STARTING.equals(ObjectUtils.getState(instance))) {
+                    try {
+                        getObjectProcessManager().scheduleProcessInstance(InstanceConstants.PROCESS_CREATE, instance, makeData());
+                    } catch (ProcessCancelException e) {
+                        // TODO Figure out if i need this catch
+                    }
+                }
+            }
+        });
+    }
+
+    public static boolean isNativeDockerStart(ProcessState state) {
+        Boolean nativeDockerStart = DataAccessor.fromMap(state.getData()).withScope(ContainerEventCreate.class)
+                .withKey(InstanceConstants.DATA_NATIVE_DOCKER_START).as(Boolean.class);
+
+        return nativeDockerStart == null ? false : nativeDockerStart;
+    }
+
+    protected Map<String, Object> makeData() {
+        Map<String, Object> data = new HashMap<String, Object>();
+        DataAccessor.fromMap(data).withScope(ContainerEventCreate.class).withKey(InstanceConstants.DATA_NATIVE_DOCKER_START).set(true);
+        return data;
+    }
+
+    void setHost(ContainerEvent event, Instance instance) {
+        DataAccessor.fields(instance).withKey(InstanceConstants.FIELD_REQUESTED_HOST_ID)
+                .set(event.getHostId());
+    }
+
+    void setName(Map<String, Object> inspect, Instance instance) {
+        String name = (String)inspect.get(INSPECT_NAME);
+        name = name.replaceFirst("/", "");
+        instance.setName(name);
+    }
+
+    void setNetwork(Long accountId, Map<String, Object> inspect, Instance instance) {
+        String networkKind = NetworkConstants.KIND_NETWORK;
+        if ( BooleanUtils.toBoolean(getRancherNetworkLabel(inspect)) ) {
+            networkKind = NetworkConstants.KIND_HOSTONLY;
+        }
+
+        List<? extends Network> networks = networkDao.getNetworksForAccount(accountId, networkKind);
```

Comment:
Going to use it, but looks like I'll need to change the sig to `Network getNetworkForObject(Object object, String kind)`